src/transformers/models/clap/modeling_clap.py

```diff
@@ -0,0 +1,2408 @@
+            heads.
+    """
+
+    text_embeds: Optional[torch.FloatTensor] = None
+    last_hidden_state: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+@dataclass
+class ClapAudioModelOutput(ModelOutput):
+    """
+    ClapAudio model output to mimic the output of the original implementation.
+
+    Args:
+        framewise_output (`torch.FloatTensor` of shape `(batch_size, reshaped_hidden_size, num_classes)`):
+            Output hidden_states that are interpolated after applying sigmoid. These logits are used to compute the the
+            classification label in the original implementation.
+        clipwise_output (`torch.FloatTensor` of shape `(batch_size, hidden_size, num_classes)`):
+            Output hidden_states after applying sigmoid. These logits are used to compute the the classification label
+            in the original implementation.
+        fine_grained_embedding (`torch.FloatTensor` of shape `(batch_size, hidden_size)`):
+            Pooled interpolated hidden_states.
+        embedding (`torch.FloatTensor` of shape `(batch_size, hidden_size)`):
+            Sequence of hidden-states at the output of the last layer of the model.
+        attentions (`tuple(torch.FloatTensor)`, *optional*, returned when `output_attentions=True` is passed or when `config.output_attentions=True`):
+            Tuple of `torch.FloatTensor` (one for each layer) of shape `(batch_size, num_heads, sequence_length,
+            sequence_length)`.
+
+            Attentions weights after the attention softmax, used to compute the weighted average in the self-attention
+            heads.
+        hidden_states (`tuple(torch.FloatTensor)`, *optional*, returned when `output_hidden_states=True` is passed or when `config.output_hidden_states=True`):
+            Tuple of `torch.FloatTensor` (one for the output of the embeddings, if the model has an embedding layer, +
+            one for the output of each layer) of shape `(batch_size, sequence_length, hidden_size)`.
+
+            Hidden-states of the model at the output of each layer plus the optional initial embedding outputs.
+    """
+
+    framewise_output: torch.FloatTensor = None
+    clipwise_output: torch.FloatTensor = None
+    fine_grained_embedding: torch.FloatTensor = None
+    embedding: torch.FloatTensor = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+
+
+@dataclass
+class ClapAudioModelOutputWithProjection(ModelOutput):
+    """
+    ClapAudio model output to mimic the output of the original implementation.
+
+    Args:
+        audio_embeds (`torch.FloatTensor` of shape `(batch_size, output_dim)`):
+            The audio embeddings obtained by applying the projection layer to the pooler_output.
+        framewise_output (`torch.FloatTensor` of shape `(batch_size, reshaped_hidden_size, num_classes)`):
```

Comment:
We probably don't need that. If the previous class has audio_embeds 😉

Reply:
+1! See https://github.com/huggingface/transformers/pull/21370#discussion_r1102591650